typed-racket-lib/typed-racket/typecheck/tc-app-helper.rkt

```diff
@@ -124,12 +128,17 @@
                   (format "Wrong number of arguments - Expected ~a, but got ~a\n\n" (length (car doms)) (length arg-tys))
                   "")
               (append
-               (for/list ([dom-t (in-list (list-extend arg-tys (car doms) #f))]
+               (for/list ([i (in-naturals 1)]
+                          [dom-t (in-list (list-extend arg-tys (car doms) #f))]
                           [arg-t (in-list (list-extend (car doms) arg-tys #f))]
-                          [i (in-naturals 1)])
+                          [arg-id (in-list/rest arg-names #f)])
                          (let ([dom-t (or dom-t "-none-")]
                                [arg-t (or arg-t "-none-")])
-                           (format "Argument ~a:\n  Expected: ~a\n  Given:    ~a\n" i (make-printable dom-t) (make-printable arg-t))))
+                           (format "Argument ~a~a:\n  Expected: ~a\n  Given:    ~a\n"
+                                   i
+                                   (if arg-id (format " (~a)" (syntax-e arg-id)) "")
```

Comment:
minor: if `arg-id`, I think it'd be better to print "Argument x (position i): ....."

Reply:
agreed - done